src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,188 @@
+        taskList = Storage.readList();
+    }
+
+    /**
+     * Creates a task and adds it to the task list: todo
+     *
+     * @param type     task type
+     * @param taskName task name
+     * @return a string representation of the added task
+     * @throws IOException if error occurs in file accessing
+     */
+    public static String add(TaskType type, String taskName) throws IOException {
+        try {
+            getTaskListSize();
+            Todo todo = new Todo(taskName);
+            taskList.add(todo);
+            Storage.addToList(TaskType.TODO, 0, taskName, "");
+            return todo.toString();
+        } catch (IOException e) {
+            throw new IOException("File access failed");
+        }
+    }
+
+    /**
+     * Creates a task and adds it to the task list: deadline or event
+     *
+     * @param type     task type
+     * @param taskName task name
+     * @param time     time of the task
+     * @return a string representation of the added task
+     * @throws IOException if error occurs in file accessing
+     */
+    public static String add(TaskType type, String taskName, String time) throws IOException {
+        try {
+            getTaskListSize();
+            if (type == TaskType.DEADLINE) {
+                Deadline deadline = new Deadline(taskName, time);
+                taskList.add(deadline);
+                Storage.addToList(TaskType.DEADLINE, 0, taskName, time);
+                return deadline.toString();
+            } else {
+                Event event = new Event(taskName, time);
+                taskList.add(event);
+                Storage.addToList(TaskType.EVENT, 0, taskName, time);
+                return event.toString();
+            }
+        } catch (IOException e) {
+            throw new IOException("File access failed");
+        }
+
+    }
+
+
+    /**
+     * Finds tasks that contain a given keyword
+     *
+     * @param word keyword of the task
+     * @return a string representation of the tasks found
+     * @throws IOException               if error occurs in file accessing
+     * @throws InvalidParameterException if word is blank
+     */
+    public static String findTask(String word)
+            throws IOException, InvalidParameterException, NullListException {
+        if (word.isBlank()) {
+            throw new InvalidParameterException("Invalid input");
+        }
+        int size = getTaskListSize();
+        if (size == 0) {
+             throw new NullListException();
+        }
+        StringBuilder sb = new StringBuilder();
+        int count = 1;
+        boolean isFound = false;
+        for (int i = 1; i <= size; i++) {
+            String task = taskList.get(i - 1).toString();
+            if (task.contains(word)) {
+                sb.append("\n").append(count).append(". ").append(task);
+                isFound = true;
+                count++;
+            }
+        }
+        if (!isFound) {
```

Comment:
The boolean isFound might be redundant since the toString() method for string builder will return an empty String if nothing is appended.